pipeline/metadata/satellite.py

```diff
@@ -341,8 +341,7 @@ def _key_by_date_domain(row: Row) -> Tuple[DateDomainKey, Row]:
 def _partition_test_and_control(row_tuple: Tuple[DateDomainKey, Row],
                                 _: int) -> int:
   row: Row = row_tuple[1]
-  # 'anomaly' is None for control measurements
-  if row['is_control_ip'] or row['anomaly'] is None:
+  if row['is_control_ip'] or row['is_control']:
```

Comment:
The measurements for control domains (`row['is_control']`) should not be in this pcollection. These could be partitioned in a separate function if they are causing the memory bug.

Reply:
I think you could first partition the rows based on the domain (test domain vs. control domain), then run the existing steps on the measurements for test domains (partition test vs. control resolver, etc.)

Reply:
I've split this into three pieces. controls that are both domain+ip controls are going into the domain controls pcollection.